### PR TITLE
Config file handling

### PR DESCRIPTION
**Pull request by fvox13**

### CODE REVIEW

**Comment by kruchone** [comment]
Looks like this fix has changes from the other PR. Could you revert the config file changes from this one?

**Comment by fvox13** [comment]
I think I may have done the PR incorrectly.  Is it possible to just exclude the first commit when you merge?

**Comment by kruchone** [comment]
Potentially doing a `git revert 9ec8562` and `git revert c355208` on that branch would undo the two commits. I can't even merge it because it is conflicting with the last PR that was merged (because we would be merging the same one twice). Since this is on your fork, I would have to check out your branch on your fork and modify it. Let me know how you want to proceed, I can commit into your branch if it works better for you.

Also, if you could make the spacing consistent with jambi.py in config.py that would be good (although at some point I will re-run flake8 and fix spacing anyway).

File: jambi/config.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+				"{} environment variable is set, but no file was found" \
+				.format(ENVIRONMENT_VARIABLE)
+			)
+	else:
+		if os.path.isfile(DEFAULT_CONFIG_FILENAME):
+			return DEFAULT_CONFIG_FILENAME
+		else:
+			raise ImproperlyConfigured(
+				"No configuration file found. Either create {} in the " \
+				"install directory, or set the {} environment variable." \
```
Comment (kruchone) [inline jambi/config.py:24]: Potentially confusing to people who might not know what the 'install directory' is. Maybe alternative wording would be something like. "[...]Either create jambi.conf in _this_ directory[...]"

What do you think?

Edit: In fact, we could just remove the bit "in the install directory"

File: jambi/config.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import os
+
+from jambi.exceptions import ImproperlyConfigured
+
+ENVIRONMENT_VARIABLE = "JAMBI_CONFIG"
```
Comment (kruchone) [inline jambi/config.py:5]: Is the value in this ENV variable relative or absolute? I can see usage either way, but was leaning toward absolute path. What are your thoughts?

**Review by kruchone — CHANGES_REQUESTED** [review]

**Comment by kruchone** [comment]
I would like to see an extra `argparse` option `--config` that allows someone to point to a [relative or absolute] path to a config file to use as `jambi.conf`, even if it isn't named `jambi.conf`.

If you want that to be handled separately I am happy to do it, just thought I would mention here as we are dealing with configgy things :)

File: jambi/config.py (hunk 2 above)
Comment (fvox13) [reply jambi/config.py:5]: Absolute.  It just gets passed to the config parser as-is, assuming we can verify it's a file.

File: jambi/config.py (hunk 1 above)
Comment (fvox13) [reply jambi/config.py:24]: Yeah, it's also kind of weird to put a config file in your site-packages directory, now that I think through it...

**Comment by fvox13** [comment]
Recommend closing this PR in favor of #12

**Pull request closed without merging**